frontend/apps/web/src/pages/system/_name.tsx

```diff
@@ -1,65 +1,68 @@
 import { PageContainer } from '@/containers/PageContainer'
 import {
-  BoxLayout,
-  CardLayout,
-  Colors,
-  ColumnLayout,
-  Icon,
-  MatrixMap,
   PageLayout,
   StackLayout,
   Text,
+  Icon,
   usePageLayout,
 } from '@filecoin/ui'
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
 import { useTranslation } from 'react-i18next'
-import { NavLink, useParams } from 'react-router-dom'
-import ReactTooltip from 'react-tooltip'
-import styled from 'styled-components'
+import { useParams, useNavigate } from 'react-router-dom'
+import { DetailedView } from '@/components/system/DetailedView'
+import { Overview } from '@/components/system/Overview'
+
+const TABS = ['overview', 'detailedView']
 
 const RepositoryDetails = () => {
   const {
     state: { model },
   } = PageContainer.useContainer()
   const params: { name: string } = useParams()
   const { t } = useTranslation()
+  const navigate = useNavigate()
   const system = model.findSystemByName(params.name)
-  const testKinds = model.getAllTestKinds()
   const totalTests = system.subsystems.reduce((totalTests, subsystem) => {
     return totalTests + subsystem.tests.length
   }, 0)
-  const [activeTab, setActiveTab] = useState(1)
 
-  const Header = props => {
-    return (
-      <PageLayout.Header>
-        <Text type="heading 5">Systems</Text>
+  const [activeTab, setActiveTab] = useState<'overview' | 'detailedView'>(
+    'overview',
+  )
 
+  useEffect(() => {
+    if (!TABS.includes(activeTab)) {
+      setActiveTab('overview')
+    }
+    navigate({
```

Comment:
we can add navigate to else {} because we don't won't to navigate if the queryParam is wrong.